src/Rule/Ip.php

```diff
@@ -222,165 +204,142 @@ public function __construct()
      * Example:
      *
      * ```php
-     * [
-     *      'ranges' => [
+     * (new Ip())->ranges([
      *          '192.168.10.128'
      *          '!192.168.10.0/24',
      *          'any' // allows any other IP addresses
-     *      ]
-     * ]
+     *      ]);
      * ```
      *
      * In this example, access is allowed for all the IPv4 and IPv6 addresses excluding the `192.168.10.0/24` subnet.
      * IPv4 address `192.168.10.128` is also allowed, because it is listed before the restriction.
-     * @property array the IPv4 or IPv6 ranges that are allowed or forbidden.
-     * See [[setRanges()]] for detailed description.
+     * @return static
      */
-    public function setRanges($ranges)
+    public function ranges(array $ranges)
```

Comment:
```suggestion
    public function ranges(array $ranges): self
```

Reply:
Phpdoc `@return static` is a better solution because `self` must be compliance with the parent class name after inheritance.

example
```php
MyExtendendIp extends Ip {         ˇˇ----- the PHP processor throws an error if I do not use the parent class.
  public function ranges($ranges): Ip {
     // overerride
  }
}
```

Of course, this is not a problem if we set the `Ip` class to `final`.